pkg/reader/mysql/reader.go

```diff
@@ -14,83 +14,62 @@ import (
 
 // Storage ...
 type storage struct {
-	generic.SqlReader
-
-	tables []string
+	connection *sql.DB
 }
 
 // NewStorage ...
 func NewStorage(conn *sql.DB) reader.Reader {
-	return &storage{
-		SqlReader: generic.SqlReader{
-			Connection: conn,
-			QuoteIdentifier: func(name string) string {
-				return "`" + strings.Replace(name, "`", "``", -1) + "`"
-			},
-		},
-	}
+	return generic.NewSqlReader(conn, &storage{connection: conn})
```

Comment:
This seems a bit weird. The SQLReader owns the `connection` obj, and `storage` also owns the connection obj but the Reader receives a storage... It seems a bit off. Wdyt?

Reply:
I know but I don't really know a better way there is no inheritance in go so i'm kinda lost in my OOP brain any suggestion on how to improve this? add `getConnection` to the SqlEngine interface?